src/main/java/net/imglib2/roi/util/IterableRegionOnBooleanRAI.java

```diff
@@ -72,55 +74,70 @@ public IterableRegionOnBooleanRAI( final RandomAccessibleInterval< T > interval,
 		super( interval );
 		this.size = size;
 		sourceIterable = Views.iterable( interval );
+		inside = new InsideIterable();
 	}
 
 	@Override
-	public long size()
+	public RandomAccess< T > randomAccess()
 	{
-		return size;
+		return sourceInterval.randomAccess();
 	}
 
 	@Override
-	public Void firstElement()
+	public RandomAccess< T > randomAccess( final Interval interval )
 	{
-		if ( size() == 0 )
-			throw new NoSuchElementException();
-		return cursor().next();
+		return sourceInterval.randomAccess( interval );
 	}
 
 	@Override
-	public Object iterationOrder()
+	public IterableInterval< Void > inside()
 	{
-		return this;
+		return inside;
 	}
 
-	@Override
-	public Iterator< Void > iterator()
+	class InsideIterable extends AbstractWrappedInterval< Interval > implements IterableInterval< Void >
```

Comment:
Make these inner `InsideIterable` classes private? Just for better encapsulation.

Reply:
Same goes for `PositionableInsideCursor`

Reply:
Good idea!